tar/storage/storage_transaction.c

```diff
@@ -327,16 +327,26 @@ storage_transaction_start(NETPACKET_CONNECTION * NPC, uint64_t machinenum,
 		warn0("Sequence number mismatch: Run --fsck");
 		goto err0;
 	case 2:
-		/* We should only get this for a write transaction start. */
```

Comment:
I'm not certain what you were thinking at the time, so I don't know if this is a decent fix.

The previous significant changes to this file were in 2010-06-27 for tarsnap 1.0.27 -- that added the `fsck` with `whichkey` being 2 or 3, for example.  This particular message was introduced in 2008-10-14 for tarsnap 1.0.14.

Reply:
oh, in case it helps, here's some git hashes:

1.0.14: fe41096be6d1ec38ea35a175f3d82605e104b604
1.0.27: 04fdb829bb369ad099e3b5bfe35086e0cb93f2c9

Reply:
Yeah, I was right that this had to do with using the write keys for fsck starting at 1.0.27.  The case of using the delete key for fsck is fine already (we'll never get ENOMONEY in that case).